packages/react-components/react-nav-preview/src/components/NavCategoryItem/useNavCategoryItem.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import { getIntrinsicElementProps, mergeCallbacks, slot, useEventCallback } from '@fluentui/react-utilities';
+import { ChevronRightRegular } from '@fluentui/react-icons';
+import { NavCategoryItemProps, NavCategoryItemState } from './NavCategoryItem.types';
+import { useNavCategoryContext_unstable } from '../NavCategoryContext';
+import { useNavContext_unstable } from '../NavContext';
+
+/**
+ * Create the state required to render NavCategoryItem.
+ *
+ * The returned state can be modified with hooks such as useNavCategoryItemStyles,
+ * before being passed to renderNavCategoryItem.
+ *
+ * @param props - props from this instance of NavCategoryItem
+ * @param ref - reference to root HTMLButtonElement of NavCategoryItem
+ */
+export const useNavCategoryItem_unstable = (
+  props: NavCategoryItemProps,
+  ref: React.Ref<HTMLButtonElement>,
+): NavCategoryItemState => {
+  const { onClick, expandIcon } = props;
+
+  const { open, value } = useNavCategoryContext_unstable();
+
+  const { onRequestNavCategoryItemToggle } = useNavContext_unstable();
+
+  const onNavCategoryItemClick = useEventCallback(
+    mergeCallbacks(onClick, event => onRequestNavCategoryItemToggle(event, { type: 'click', event, value })),
+  );
+
+  const expandIconRotation = open ? -90 : 90;
+
+  // TODO - these are copied from AccordionHeader.
+  // We need to figure out if they are applicable to this
+  // scenario and adapt them accordingly.
+
+  // const buttonSlot = slot.always(button, {
+  //   elementType: 'button',
+  //   defaultProps: {
+  //     // we may decide to light these up later
+  //     // disabled,
+  //     // disabledFocusable,
+  //     'aria-expanded': open,
+  //     type: 'button',
+  //     onClick: onNavCategoryItemClick,
+  //   },
+  // });
+
+  // buttonSlot.onClick = useEventCallback(event => {
+  //   if (isResolvedShorthand(button)) {
+  //     button.onClick?.(event);
+  //   }
+  //   if (!event.defaultPrevented) {
+  //     onRequestNavCategoryItemToggle(event, { value, event }); //({ value, event });
+  //   }
+  // });
+
+  return {
+    open,
+    value,
+    // TODO add appropriate props/defaults
+    components: {
+      root: 'button',
+      // button: 'div',
+      expandIcon: 'span',
+      // icon: 'div',
+    },
+    root: slot.always(
+      getIntrinsicElementProps('button', {
+        ref,
+        role: 'nav',
+        type: 'navigation',
+        ...props,
+        onClick: onNavCategoryItemClick,
+      }),
+      { elementType: 'button' },
+    ),
+    expandIcon: slot.always(expandIcon, {
+      defaultProps: {
+        children: <ChevronRightRegular style={{ transform: `rotate(${expandIconRotation}deg)` }} />,
```

Comment:
Instead of setting `style` on the icon, instead add a class to the `expandIcon` slot that sets a rotation. E.g.

```ts
const useExpandIconStyles = makeStyles(
  open: {
    transform: 'rotate(-90deg)',
  },
  closed: {
    transform: 'rotate(90deg)',
  },
});
```

Reply:
Bonus points: pick a different Chevron icon that doesn't need to be rotated in the closed state (`ChevronDownRegular`?)

Reply:
yeah, I like that way better.

I didn't focus much on the styles for this PR, but every little bit helps :-)

Reply:
cool - got the styles applied properly, and the state hook tidied. Icon picking is a problem for a future PR 😂